### PR TITLE
added inherited virtual game wrapper

### PR DESCRIPTION
**Pull request by lourw**
# Description

- added an inheritedWidget to the root of the project. This widget provides a `VirtualPetGame` instance that child nodes within the app can manipulate
- added a testing page that demonstrates the ability of a page to interact with the `VirtualPetGame`. There is a start and pause button that will play and pause the game's execution.
- removed a test file that was blocking this pr, the test was irrelevant to the running of the app, but a new version that works with inhirtedwidgets will be created
- renamed `MyApp` to `VirtualPetApp`

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

N/A this was a app design change, no logic required

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wu-jacob — APPROVED** [review]

File: front-end/virtual_pet/lib/inherited_widgets/inherited_virtual_pet_game_wrapper.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+import 'package:virtual_pet/flame/game/virtual_pet_game.dart';
+
+class InheritedVirtualPetGameWrapper extends InheritedWidget {
+  final VirtualPetGame virtualPetGame;
+
+  const InheritedVirtualPetGameWrapper({
+    required this.virtualPetGame, 
+    required Widget child, 
+    super.key}) : super(child: child);
```
Comment (mebesius) [inline front-end/virtual_pet/lib/inherited_widgets/inherited_virtual_pet_game_wrapper.dart]: Please bring the `})` down to a new line

File: front-end/virtual_pet/lib/inherited_widgets/inherited_virtual_pet_game_wrapper.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  @override 
+  bool updateShouldNotify(InheritedWidget oldWidget) => true;
```
Comment (mebesius) [inline front-end/virtual_pet/lib/inherited_widgets/inherited_virtual_pet_game_wrapper.dart]: Do we want this to always return true?

**Review by mebesius — CHANGES_REQUESTED** [review]

File: front-end/virtual_pet/lib/inherited_widgets/inherited_virtual_pet_game_wrapper.dart (hunk 2 above)
Comment (lourw) [reply front-end/virtual_pet/lib/inherited_widgets/inherited_virtual_pet_game_wrapper.dart]: fixed

**Comment by lourw** [comment]
@mebesius made changes and also rebased eric's changes in

**Review by mebesius — APPROVED** [review]
Thanks for making those changes @lou-weng!

**Pull request merged**